fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarks.kt

```diff
@@ -180,23 +180,7 @@ private fun RecentBookmarkImage(bookmark: RecentBookmark) {
                     PlaceholderBookmarkImage()
                 }
 
-                WithIcon { icon ->
-                    Box(
-                        modifier = imageModifier.background(
-                            color = FirefoxTheme.colors.layer2,
-                        ),
-                        contentAlignment = Alignment.Center,
-                    ) {
-                        Image(
-                            painter = icon.painter,
-                            contentDescription = null,
-                            modifier = Modifier
-                                .size(36.dp)
-                                .clip(cardShape),
-                            contentScale = ContentScale.Crop,
-                        )
-                    }
-                }
+                FallbackBookmarkFaviconImage(bookmark.url)
```

Comment:
Let's do the if check here instead of inside of `FallbackBookmarkFaviconImage`.
```suggestion
         if (!bookmark.url.isNullOrEmpty()) {
                FallbackBookmarkFaviconImage(bookmark.url)
         }
```